src/syntaxtree.c

```diff
@@ -272,6 +422,50 @@ static Obj SyntaxTreeIf(Obj result, Stat stat)
     return result;
 }
 
+static Expr SyntaxTreeCodeIf(Obj node)
+{
+    Obj   branches = ELM_REC(node, RNamName("branches"));
+    UInt  len = LEN_PLIST(branches);
+    UInt1 tnum = FIND_TNUM_FROM_STRING(ELM_REC(node, RNamName("type")));
+    Expr  ifexpr = NewStatWithProf(tnum, 2 * len * sizeof(Expr), 0);
+    for (int i = 0; i < len; i++) {
+        Obj  condbodypair = ELM_PLIST(branches, i + 1);
+        Obj  conditionnode = ELM_REC(condbodypair, RNamName("condition"));
+        Obj  bodynode = ELM_REC(condbodypair, RNamName("body"));
+        Expr condition = SyntaxTreeDefaultCoder(conditionnode);
+        Expr body = SyntaxTreeDefaultCoder(bodynode);
+        WRITE_EXPR(ifexpr, 2 * i, condition);
+        WRITE_EXPR(ifexpr, 2 * i + 1, body);
+    }
+    return ifexpr;
+}
+
+static Expr SyntaxTreeCodeImmediateInteger(Obj node)
+{
+    Obj value = ELM_REC(node, RNamName("value"));
+    return INTEXPR_INT(Int_ObjInt(value));
```

Comment:
So yeah, this is indeed unsafe because the resulting Int might be too big. So I'd switch back from `Int_ObjInt ` to `INT_INTOBJ`, and instead insert an `IS_INTOBJ` check before (resp. use `RequireSmallInt`).

In an ideal world, though, I'd unify handling of `T_INTEXPR` and `T_INT_EXPR`: use only of them in the generated prec "tree"; then when coding  back into a function, look at the value, and depending on whether it is immediate or not, output a `T_INTEXPR` or `T_INT_EXPR`... That would make it nicer to use. But doesn't have to happen in this PR!